sklearn/model_selection/_search.py

```diff
@@ -918,15 +1005,32 @@ class RandomizedSearchCV(BaseSearchCV):
 
     Attributes
     ----------
-    grid_scores_ : list of named tuples
-        Contains scores for all parameter combinations in param_grid.
-        Each entry corresponds to one parameter setting.
-        Each named tuple has the attributes:
-
-            * ``parameters``, a dict of parameter settings
-            * ``mean_validation_score``, the mean score over the
-              cross-validation folds
-            * ``cv_validation_scores``, the list of scores for each fold
+    search_results_ : dict of numpy (masked) ndarrays
```

Comment:
Rather, this *sort* of change has been proposed and sought by the core devs for years; the fact that model_selection has not been released maybe makes it the best time to do it. However, I think we **must** consider this a deprecation case, not just drop `grid_scores_` altogether. And that's because users will initially *not* rewrite all their old code but will change some imports, and will be much less angry if they get a `DeprecationWarning` at the end of a long fit rather than an `AttributeError`